src/cascadia/Remoting/Peasant.cpp

```diff
@@ -289,4 +290,24 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
                           TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
                           TraceLoggingKeyword(TIL_KEYWORD_TRACE));
     }
+
+    // Method Description:
+    // - Request and return the window layout from the current TerminalPage
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - the window layout as a json string
+    hstring Peasant::GetWindowLayout()
+    {
+        auto args = winrt::make_self<implementation::GetWindowLayoutArgs>();
+        _GetWindowLayoutRequestedHandlers(nullptr, *args);
+        if (const auto op = args->WindowLayoutJsonAsync())

```

Comment:
this is wild:
* the peasant asks for window layouts (on a COM bg thread)
* The apphost gets it's inquiry
  * hops over to the UI thread,
  * gets the layout
  * hops back to the COM thread
  * returns the layout
* the peasant returns the string on that COM thread

then in the totally other process, the monarch collects these all up, without even thinking about it. _COM is magic_